File: benchmarks/collatz.c
#include <stdio.h>
static long collatz(long n) {
    long c = 0;
    while (n > 1) { n = (n & 1) ? n*3+1 : n/2; c++; }
    return c;
}
int main(void) {
    long total = 0;
    for (long i = 1; i <= 200000; ++i) total += collatz(i);
    printf("total=%ld\n", total);
    return 0;
}

File: benchmarks/collatz.rs
fn collatz(mut n: i64) -> i64 {
    let mut c = 0i64;
    while n > 1 { n = if n & 1 == 0 { n / 2 } else { n*3+1 }; c += 1; }
    c
}
fn main() {
    let mut total: i64 = 0;
    for i in 1..=200_000i64 { total += collatz(i); }
    println!("total={}", total);
}
